subsys/bluetooth/controller/crypto/crypto.c

```diff
@@ -4,33 +4,33 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <soc.h>
+#include <drivers/entropy.h>
 
 #define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
 #define LOG_MODULE_NAME bt_ctlr_crypto
 #include "common/log.h"
 
 #include "hal/ecb.h"
 
+static struct device *entropy_driver;
+
 int bt_rand(void *buf, size_t len)
 {
-	u8_t *buf8 = buf;
-
-	while (len) {
-		u32_t v = sys_rand32_get();
-
-		if (len >= sizeof(v)) {
-			memcpy(buf8, &v, sizeof(v));
-
-			buf8 += sizeof(v);
-			len -= sizeof(v);
-		} else {
-			memcpy(buf8, &v, len);
-			break;
-		}
+	struct device *dev = entropy_driver;
```

Comment:
do you need the local variable?

Reply:
Technically no. I basically copy/pasted that part from rand32_entropy_device.c